src/client/pages/Offer/Introduction/Sidebar/useHasScrolledPastElement.ts

```diff
@@ -0,0 +1,68 @@
+import React, { useCallback, useEffect, useState } from 'react'
+import { throttle } from 'utils/throttle'
+
+type StickyOptions = {
+  /**
+   * The margin that should be used to to calculate the full element height + paddings + margins
+   **/
+  margin?: number
```

Comment:
I'm thinking that we could extend this to `marginTop` and `marginBottom` if we see the need in the future